lib/parser.ts

```diff
@@ -56,9 +58,7 @@ export class Parser {
         stop_price: this.parseNumber(rawOrder.stop_price),
         filled_avg_price: this.parseNumber(rawOrder.filled_avg_price),
         status: rawOrder.status as OrderStatus,
-        legs: rawOrder.legs && rawOrder.legs.length
-          ? rawOrder.legs.map(this.parseOrder)
```

Comment:
This was a bug. My bad.

The way it was written, `parseOrder` would not have the correct `this`. Since we already have a helper for handling multiple orders, it was an easier fix to use that than to use something like `bind` or a lambda here.

Glad I added these tests and found this!